llvm/test/CodeGen/X86/sadd_sat_vec.ll

```diff
@@ -861,6 +861,9 @@ define <8 x i32> @v8i32(<8 x i32> %x, <8 x i32> %y) nounwind {
 ; AVX1-NEXT:    vpcmpgtd %xmm2, %xmm3, %xmm3
 ; AVX1-NEXT:    vpcmpgtd %xmm4, %xmm0, %xmm0
 ; AVX1-NEXT:    vinsertf128 $1, %xmm3, %ymm0, %ymm0
+; AVX1-NEXT:    vcvtdq2ps %ymm1, %ymm1
+; AVX1-NEXT:    vpxor %xmm3, %xmm3, %xmm3
+; AVX1-NEXT:    vcmpltps %ymm3, %ymm1, %ymm1
```

Comment:
Not sure why this is happening. It seems we are determining the compare is dead before but not after.
Not sure what the best way to fix this is.

Reply:
Think fixed. Added support for `cmpp` in simplifydemandedbits.